CMakeLists.txt

```diff
@@ -12,8 +12,12 @@ cmake_policy(SET CMP0048 NEW) # handling project_version_* variables
 project(EnergyPlus)
 
 # Raise an error if attempting to compile on macOS older than 10.15 - it does not work
-if (APPLE AND CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS "10.15")
-  message(FATAL_ERROR "The minimum required version for macOS is 10.15, however CMAKE_OSX_DEPLOYMENT_TARGET is set to ${CMAKE_OSX_DEPLOYMENT_TARGET}. Please set CMAKE_OSX_DEPLOYMENT_TARGET to 10.15 or greater and try again.")
+if (APPLE)
+  if ("${CMAKE_OSX_DEPLOYMENT_TARGET}" STREQUAL "")
+    message(FATAL_ERROR "CMAKE_OSX_DEPLOYMENT_TARGET not set. Please set CMAKE_OSX_DEPLOYMENT_TARGET to 10.15 or greater and try again.")
+  elseif (CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS "10.15")
+    message(FATAL_ERROR "The minimum required version for macOS is 10.15, however CMAKE_OSX_DEPLOYMENT_TARGET is set to ${CMAKE_OSX_DEPLOYMENT_TARGET}.")
+  endif()
```

Comment:
I find it strange that it would be empty. You've run into this? What's your cmake version?

From the docs (https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html)

> If not set explicitly the value is initialized by the `MACOSX_DEPLOYMENT_TARGET` environment variable, if set, and **otherwise computed based on the host platform.**

Reply:
```
$ cmake --version
cmake version 3.20.5
```

Reply:
Running `ccmake ..` in a fresh build folder on develop, I get the following:

```
The C compiler identification is AppleClang 14.0.3.14030022
 The CXX compiler identification is AppleClang 14.0.3.14030022
 Detecting C compiler ABI info
 Detecting C compiler ABI info - done
 Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
 Detecting C compile features
 Detecting C compile features - done
 Detecting CXX compiler ABI info
 Detecting CXX compiler ABI info - done
 Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
 Detecting CXX compile features
 Detecting CXX compile features - done
 CMake Error at CMakeLists.txt:16 (message):
   The minimum required version for macOS is 10.15, however
   CMAKE_OSX_DEPLOYMENT_TARGET is set to .  Please set
   CMAKE_OSX_DEPLOYMENT_TARGET to 10.15 or greater and try again.

 Configuring incomplete, errors occurred!
 See also "/Users/mmitchel/Projects/EnergyPlus/dev/develop/build/CMakeFiles/CMakeOutput.log".
 See also "/Users/mmitchel/Projects/EnergyPlus/dev/develop/build/CMakeFiles/CMakeError.log".
```

Reply:
Ok I can reproduce when I don't use my bash_profile, which contains this...:

```
    # Set architecture flags
    export ARCHFLAGS="-arch arm64"
    export MACOSX_DEPLOYMENT_TARGET=12.1
    export SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk
```

The CMake documentation is misleading. Can't say I'm shocked there...

Thanks for the clarification @mitchute